.github/workflows/unit-tests.yml

```diff
@@ -1,10 +1,10 @@
 name: unit-tests
 
 on:
-  workflow_dispatch:
   push:
+    branches: [ "main" ]
   pull_request:
-    types: [ opened, edited, reopened, synchronize ]
```

Comment:
This setup meant the PRs were run twice for every push to a branch. The new setup means unit tests will be run on pull requests (all pushes) then once again when they are merged into the main branch,

Reply:
Nice spot and modification will add it to the other repo's with tests as well then